cdap-hbase-compat-0.96/src/main/java/co/cask/cdap/data2/util/hbase/HBase96TableUtil.java

```diff
@@ -38,6 +46,60 @@
  *
  */
 public class HBase96TableUtil extends HBaseTableUtil {
+
+  @Override
+  public boolean namespacesSupported() {
+    return true;
+  }
+
+  @Override
+  public HTable getHTable(Configuration conf, TableId tableId) throws IOException {
+    Preconditions.checkArgument(tableId != null, "Table id should not be null");
+    return new HTable(conf, TableName.valueOf(toHBaseNamespace(tableId.getNamespace()), tableId.getTableName()));
+  }
+
+  @Override
+  public HTableDescriptor getHTableDescriptor(TableId tableId) {
+    Preconditions.checkArgument(tableId != null, "Table id should not be null");
+    return new HTableDescriptor(TableName.valueOf(toHBaseNamespace(tableId.getNamespace()), tableId.getTableName()));
+  }
+
+  @Override
+  public boolean hasNamespace(HBaseAdmin admin, Id.Namespace namespace) {
+    Preconditions.checkArgument(admin != null, "HBaseAdmin should not be null");
+    Preconditions.checkArgument(namespace != null, "Namespace should not be null.");
+    try {
+      admin.getNamespaceDescriptor(toHBaseNamespace(namespace));
+      return true;
+    } catch (IOException e) {
```

Comment:
This should only catch NamespaceNotFoundException.  Other IOExceptions may be due to other conditions and do not indicate the presence or absence of a namespace.